### PR TITLE
Fix D_inv and cg of DPGD: warm start, using inversion for non convolution case

### PR DESCRIPTION
**Pull request by EnLAI111**

### CODE REVIEW

File: solvers/DPGD.py [hunk 1]
```diff
@@ -34,19 +34,27 @@ def set_objective(self, A, reg, y, c, delta, data_fit):
...
-        L = np.tri(p)[:, 1:]
-        DA_inv = np.linalg.pinv(self.A @ L)
+        if isinstance(self.A, np.ndarray):
+            DA_inv = np.diff(np.linalg.pinv(self.A), axis=0)
+            DA_invDA_invt = DA_inv @ DA_inv.T
+            DA_invy = DA_inv @ self.y
+            AtA_inv = np.linalg.pinv(self.A.T @ self.A)
+        else:
+            D_inv = np.linspace(1/p - 1, - 1/p, p-1) + np.tri(p)[:, 1:]
+            DA_inv = np.linalg.pinv(self.A @ D_inv)
```
Comment (tomMoral) [inline solvers/DPGD.py:44]: Did you check this gives the right inverse? If so, could you add a test in `tests/test_matrices_operations.py`?

Comment (tomMoral) [reply solvers/DPGD.py:44]: A simple check: you should get the same results, whether you pass `A` as a ndarray or wrapped in a `LinearOperator`.
Can you check this?

Comment (EnLAI111) [reply solvers/DPGD.py:44]: There is a problem here. I will fix it.

Comment (EnLAI111) [reply solvers/DPGD.py:44]: I have check that it's the right inverse, however it works only when A is ```LinearOperator```.

Comment (tomMoral) [reply solvers/DPGD.py:44]: how did you check this is the right inverse?

Comment (EnLAI111) [reply solvers/DPGD.py:44]: '''python
def test_inverse(D, D_inv):
    n = D.shape[0]
    return np.allclose(D @ D_inv, np.identity(n))
```

**Pull request merged**